R/plot.R

```diff
@@ -0,0 +1,278 @@
+#' @importFrom ggplot2 autoplot
+#' @export
+ggplot2::autoplot
+
+#' Plot Bayesian model parameters
+#'
+#' @param object Output of \code{\link{fe_bayes_lm}}
+#' @param type A single character string for plot type with possible values
+#' "intercepts", "slopes", or "competition".
+#' @param sp_to_plot Vector of subset of species to plot
+#' @inheritParams predict.fe_bayes_lm
+#'
+#' @import ggridges
+#' @importFrom mvnfast rmvt
+#' @importFrom purrr set_names
+#' @importFrom tidyr gather
+#' @importFrom ggridges geom_density_ridges
+#' @importFrom stats as.formula
+#' @import stringr
+#' @return \code{focal_vs_comp} with new column of predicted \code{growth_hat}
+#'
+#' @examples
+#' library(ggplot2)
+#' library(ggridges)
+#'
+#' # Load in posterior parameter example
+#' data(posterior_param_ex)
+#'
+#' # Plot beta_0, growth intercepts
+#' autoplot(posterior_param_ex, type = "intercepts")
+#'
+#' # Plot beta_dbh, growth-dbh slope
+#' autoplot(posterior_param_ex, type = "slopes")
+#'
+#' # Plot lambdas, competition coefficients
+#' autoplot(posterior_param_ex, type = "competition")
+#' @export
+autoplot.fe_bayes_lm <- function(object,
```

Comment:
ah, so `autoplot.SOMETHING()` is the ggplot2 equivalent of the `plot.SOMETHING()` generic in base R. This approach is very clean!